src/util/base64-util.js

```diff
@@ -28,6 +28,21 @@ class Base64Util {
         return base64;
     }
 
+    /**
+    * Convert an array buffer to a base64 encoded string.
+    * @param {array} buffer - an array buffer to convert.
+    * @return {string} - the base64 encoded string.
+    */
+    static arrayBufferToBase64 (buffer) {
+        let binary = '';
+        const bytes = new Uint8Array(buffer);
+        const len = bytes.byteLength;
+        for (let i = 0; i < len; i++) {
+            binary += String.fromCharCode(bytes[ i ]);
+        }
+        return btoa(binary);
+    }
```

Comment:
This looks to me like it might be very slow for large buffers. Is this only used for small buffers, say under 1k? If so, I think it's fine to leave it as is and come back if it becomes a problem later. If it does get used for large buffers then I'd suggest checking it with a profiler: it could turn out that it's fine, or maybe we need to come up with a faster approach.